sqlmesh/core/context.py

```diff
@@ -478,13 +482,21 @@ def upsert_model(self, model: t.Union[str, Model], **kwargs: t.Any) -> Model:
             }
         )
 
-        update_model_schemas(self.dag, models=self._models, context_path=self.path)
+        update_model_schemas(
+            self.dag,
+            models=self._models,
+            context_path=self.path,
+            linter=self._linter,
+        )
 
         if model.dialect:
             self._all_dialects.add(model.dialect)
 
         model.validate_definition()
 
+        if self.config.linter.enabled:
+            self._linter.lint_model(model)
```

Comment:
Commenting here as well so the context is not lost:

This may result in an attribute error if you instantiate a context without loading and then try to upsert with the linter enabled.